contracts/Main.sol

```diff
@@ -62,28 +65,58 @@ contract Main {
         uint256 balance = uniswap.trbBalanceOf(msg.sender);
         require(balance > 0, "no balance to migrate");
         require(uniswap.burn(msg.sender), "burn failed");
-        newTRBContract.mint(msg.sender, balance);
-    }
-
-    // Helper function for contracts with public admin to
-    // send the funds directly to its address.
-    //slither-disable-next-line unimplemented-functions
-    function migrateContract(address _contract) public onlyAdmin {
-        address _owner = Owned(_contract).owner();
-        migrateContractTo(_contract, _owner);
+        newTRBContract.migrateAddress(msg.sender, balance);
     }
 
     //slither-disable-next-line unimplemented-functions
     function migrateContractTo(address _contract, address _owner)
         public
         onlyAdmin
     {
+        _migrateContractTo(_contract, _owner);
+    }
+
+    function migrateContractToBatch(
```

Comment:
Yeah I was thinking about something like this as well. Thanks.
Shall we use this even in the constructor?

Reply:
Doing in the constructor doesn't save that much gas,  so it's more of  a personal choice.